### ISSUE TITLE
Schedule lights : Add transition time

### ISSUE THREAD

**Issue by trantoriana** [nielsfaber/scheduler-card#700]
### Checklist

- [X] I checked for similar existing requests (both open and closed) before posting.
- [X] My request is generic, other users may benefit from it too.
- [X] I will participate in further discussion about this feature and can test it (if requested) once it's done.

### Proposal

It would be beneficial to add an optional 'transition time' of a light to a scheduled LIGHT ON event, similar to the optional 'brightness' setting.

(Not sure if the card needs to be adapted only or also the scheduler component.

### Additional info

Add optional transition time below optional brigtness option.
Values for optional transition should be at least 1 hour to be usefull in most situations (mine is aquarium light control, so longer times are preferred).

**Comment by nielsfaber** [comment]
You can add a custom action to the card for this. See https://github.com/nielsfaber/scheduler-card#actions for details.

Here is an example that should (I didnt test it) give you a slider bar with transition time.
```yaml
customize:
  light:
    actions:
      - service: turn_on
        variables:
          transition:
            name: "Transition time in seconds"
            min: 1
            max: 3600
            optional: true
          brightness:
            name: "Brightness"
            min: 0
            max: 255
            optional: true
        name: "Turn on with transition"
        icon: "hass:lightbulb-on-outline"
```

Since the request is highly user specific I think this should remain as a custom action and not added to the card by default.
Hope you agree.

**Comment by trantoriana** [comment]
The work around shows a side bar as part of the card indeed.
Hope it also works.

> Since the request is highly user specific I think this should remain as a custom action and not added to the card by default.
> Hope you agree.

Transition time, to me, is not more user specific as brightness as both are valid properties of light.

However, I can live with the work around over the customize.
I will see tomorrow if the added transition time in the schedule works.
If so, it is absolutely fine with me as I have to change the card only once.

Thanks, Niels.

Tran

**Comment by dnestico** [comment]
Sorry for commenting on a closed issue but Im trying to understand how this works.

I added the transition as a custom action and want to make sure how it works,  I added 1800 seconds which equals 30 minutes, does this mean the light will start to transition on 30 minutes before the selected time, so 4:30 am if the scheduled is for 5 am? 

I am trying to make it like the hue app which lets you fade lights on.

<img width="590" alt="Screenshot 2023-09-08 at 8 02 49 PM" src="https://github.com/nielsfaber/scheduler-card/assets/3034555/0b22388b-5e00-4f26-9365-d9f52acbae30">
